authors/apps/profiles/tests/test_folow.py

```diff
@@ -0,0 +1,108 @@
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from authors.apps.authentication.models import User
+from .test_data import valid_user, valid_user2
+
+class ProfileFollowAPIViewTestCase(TestCase):
+    """
+    This class defines the test suite for the view
+    that retrieves and updates a user
+    """
+
+    def setUp(self):        
+        user1 = User.objects.create_user(**valid_user)
+        user2 = User.objects.create_user(**valid_user2)
+        user2.is_active = True
+        user2.save()
+
+        self.client = APIClient()
+        self.client.force_authenticate(user=user1)
+
+
+    def test_un_authenticated_user_cannot_follow_or_unfollow_others(self):
+        self.client.logout()
+        response = self.client.post(
+            '/api/profiles/peter/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_un_authenticated_user_cannot_manage_followers(self):
+        self.client.logout()
+        response = self.client.get(
+            '/api/profiles/peter/following/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authenticated_user_cannot_follow_unknown_profile(self):
+        response = self.client.post(
+            '/api/profiles/fred/follow/'
+        )
+        self.assertIn(
+            "The requested profile does not exist.",
+            response.data["errors"]["detail"])
+
+    def test_authenticated_user_can_follow_another_user(self):
+        response = self.client.post(
+            '/api/profiles/peter/follow/'
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_authenticated_user_can_re_follow_same_user(self):
```

Comment:
Hi @reifred, thanks for the effort, this is good work. Just  a few issues to point out;
first, the name of this test is misleading. Is it a test to prove that a user **cannot** re-follow the same user or that they **can** re-follow the same user? The same goes for the method name on line 62. The assertion seems to imply that this is a negative test and yet the method implies  a positive test. Kindly clarify that. There is also a missing word in the method name on line 62.

Reply:
Thanks @richien let me update my naming. It was supposed to be cannot re-follow.